lib/services/ios-debug-service.ts

```diff
@@ -277,7 +291,41 @@ class IOSDebugService implements IDebugService {
 }
 $injector.register("iOSDebugService", IOSDebugService);
 
-function createWebSocketProxy($logger: ILogger, socketFactory: (handler: (_socket: net.Socket) => void) => void): ws.Server {
+function createTcpSocketProxy(socketFileLocation: string, $logger: ILogger, socketFactory: (handler: (socket: net.Socket) => void) => void): net.Server {
+    $logger.info("\nSetting up debugger proxy...\nPress Ctrl + C to terminate, or disconnect.\n");
+    
+    var server = net.createServer({
+        allowHalfOpen: true
+    });
+    
+    server.listen(socketFileLocation);
```

Comment:
I think this code has a potential for race condition. The server starts listening before the callback for on connection i set. Should we move this call after the server.on calls?